src/basic/format-util.c

```diff
@@ -3,23 +3,47 @@
 #include "format-util.h"
 #include "memory-util.h"
 #include "stdio-util.h"
+#include "strxcpyx.h"
 
 assert_cc(DECIMAL_STR_MAX(int) + 1 <= IF_NAMESIZE + 1);
-char *format_ifname_full(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIfnameFlag flag) {
-        /* Buffer is always cleared */
-        memzero(buf, IF_NAMESIZE + 1);
-        if (if_indextoname(ifindex, buf))
-                return buf;
+int format_ifname_full(int ifindex, FormatIfnameFlag flag, char buf[static IF_NAMESIZE + 1]) {
+        char tmp[IF_NAMESIZE + 1] = {};
+
+        if (ifindex <= 0)
+                return -EINVAL;
+
+        if (if_indextoname(ifindex, tmp)) {
+                strscpy(buf, IF_NAMESIZE, tmp);
```

Comment:
hmm, why the copy? are you saying if_indextoname() might write to the buffer on failure? really? that sucks!

Reply:
glibc uses strncpy(..., IF_NAMESIZE), so if the length is `IF_NAMESIZE`, then the buffer is not nul-terminated...

Reply:
sure, but in that case it#s enough to simply set `buf[IF_NAMESIZE] = 0` right before returning on success?

Reply:
Oh, IF_NAMESIZE includes nul at the end. So, we do not need +1. Ugh.